### PR TITLE
SOLR-16502: Multiple CopyField should not limit to first maxChars

### PR DESCRIPTION
**Pull request by risdenk**
https://issues.apache.org/jira/browse/SOLR-16502

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/update/DocumentBuilder.java [hunk 1]
```diff
@@ -336,16 +336,17 @@ private static boolean addCopyFields(
                 + ": "
                 + originalFieldValue);
       }
+      Object fieldValue = originalFieldValue;
```
Comment (cpoerschke) [inline solr/core/src/java/org/apache/solr/update/DocumentBuilder.java:339]: Perhaps `final Object originalFieldValue` could be used on the method signature to show the intention that it is not to be assigned to (any more).

**Review by cpoerschke — APPROVED** [review]

File: solr/CHANGES.txt [hunk 2]
```diff
@@ -84,6 +84,8 @@ Bug Fixes
 
 * SOLR-16274: HEAD request for managed resource returns 500 Server Error (Kevin Risden)
 
+* SOLR-16502: Multiple CopyField should not limit to first maxChars (Kevin Risden)
```
Comment (cpoerschke) [inline solr/CHANGES.txt:87]:
```suggestion
* SOLR-16502: Multiple CopyField should not limit to first maxChars (Fredrik Rodland, Kevin Risden)
```

Shared attribution based on JIRA patch attachment.

Comment (risdenk) [reply solr/CHANGES.txt]: ooo yes thanks great idea

File: solr/core/src/java/org/apache/solr/update/DocumentBuilder.java (hunk 1 above)
Comment (risdenk) [reply solr/core/src/java/org/apache/solr/update/DocumentBuilder.java:339]: fixed in 472e9a9676d3cd26f890cf3c9c6c1bfc9533f3c1

**Comment by henrik242** [comment]
@cpoerschke @risdenk Thanks! Any hope of getting this into 9.1?  I see RC1 is out now...

**Pull request merged**